src/units.rs

```diff
@@ -0,0 +1,173 @@
+use crate::utils::ConversionError;
+use std::{convert::TryFrom, fmt, str::FromStr};
+
+/// Common Ethereum unit types.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum Units {
```

Comment:
Also moved from `ethers_core`. This feels like it should live somewhere else (maybe alloy), as a primitive.

Reply:
@prestwich @DaniPopes I am OK w/ moving this to `core/` but no strong opinions